robot/impl/local_robot.go

```diff
@@ -968,6 +968,19 @@ func (r *localRobot) Reconfigure(ctx context.Context, newConfig *config.Config)
 	if err != nil {
 		allErrs = multierr.Combine(allErrs, err)
 	}
+	// Remove orphaned resources (dependents of removed resources) from newConfig.
+	for _, name := range filtered.resources.Names() {
+		for i, c := range newConfig.Components {
+			if c.ResourceName() == name {
+				newConfig.Components = append(newConfig.Components[:i], newConfig.Components[i+1:]...)
```

Comment:
I don't think order matters here so you should prefer doing something like : 
```
newConfig.Components[i]=newConfig.Components[len(...)-1]
newConfig.Components=newConfig.Components[:len(...)-1]
```

This will avoid moving things around in memory